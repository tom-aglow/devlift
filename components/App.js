import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";

import NewTodo from "./NewTodo.js";
import * as firebase from "firebase";
import firebaseConfig from "../credentials.json";

const firebaseApp = firebase.initializeApp(firebaseConfig);

class App extends Component {
  state = {
    inputValue: "",
    items: []
  };

  handleAddItem = () => {
    if (!this.state.inputValue) return null;

    const newItems = [
      ...this.state.items,
      {
        id: Date.now(),
        text: this.state.inputValue,
        completed: false
      }
    ];

    this.setState({
      items: newItems,
      inputValue: ""
    });
  };

  handleInputChange = inputValue => {
    this.setState({ inputValue });
  };

  render() {
    return (
      <View style={styles.container}>
        <Text>{"Hello world"}</Text>
        <NewTodo
          inputValue={this.state.inputValue}
          onAddItem={this.handleAddItem}
          onChange={this.handleInputChange}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  }
});

export default App;
