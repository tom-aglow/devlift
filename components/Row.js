import React, { Component } from "react";
import PropTypes from "prop-types";
import { StyleSheet, Text, View, Image } from "react-native";
import CheckBox from "react-native-check-box";

class Row extends Component {
  handleOnClick = () => {
    const { id, onToggle, isCompleted } = this.props;
    const newIsCompleted = !isCompleted;
    onToggle(id, newIsCompleted);
  };

  render() {
    const { text, isCompleted } = this.props;
    return (
      <View style={styles.container}>
        <CheckBox
          isChecked={isCompleted}
          onClick={this.handleOnClick}
          checkedImage={
            <Image source={require("../img/checkbox_checked.png")} />
          }
          unCheckedImage={
            <Image source={require("../img/checkbox_unchecked.png")} />
          }
        />
        <View style={styles.textWrapper}>
          <Text style={[styles.text, isCompleted && styles.isCompleted]}>
            {text}
          </Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    paddingVertical: 10,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between"
  },
  textWrapper: {
    flex: 1,
    marginLeft: 15
  },
  text: {
    fontSize: 16
  },
  isCompleted: {
    textDecorationLine: "line-through"
  }
});

Row.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  isCompleted: PropTypes.bool.isRequired,
  onToggle: PropTypes.func.isRequired
};

export default Row;
