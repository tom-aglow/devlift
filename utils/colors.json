{
  "white": "#FBFBFB",
  "lightGrey": "#E7E7E7",
  "mediumGrey": "#C2C2C2",
  "black": "#202020",
  "primary": "#12B4C2"
}
