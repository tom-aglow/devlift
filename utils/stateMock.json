{
  "todos": [
    { "id": 1, "text": "foo", "isCompleted": false, "listId": 0 },
    { "id": 2, "text": "bar", "isCompleted": true, "listId": 0 },
    { "id": 3, "text": "baz", "isCompleted": true, "listId": 1 }
  ],
  "lists": [
    { "id": 0, "title": "Personal" },
    { "id": 1, "title": "Movies to watch" }
  ],
  "inputValue": "",
  "editingId": 0,
  "openLists": [0]
}
