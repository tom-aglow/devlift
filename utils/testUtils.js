import React from "react";
import { mount, shallow } from "enzyme";
import { JSDOM } from "jsdom";

export const sel = id => `[data-test="${id}"]`;

export const setupGlobalObject = () => {
  const jsdom = new JSDOM("<!doctype html><html><body></body></html>");
  const { window } = jsdom;
  global.window = window;
  global.document = window.document;
};

export const TestHelper = class TestHelper {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  mountComponent(props = {}, isShallow = false) {
    const propsToUse = {
      ...this.defaultProps,
      ...props
    };

    if (isShallow) {
      return shallow(React.createElement(this.component, propsToUse));
    }

    return mount(React.createElement(this.component, propsToUse));
  }
};
