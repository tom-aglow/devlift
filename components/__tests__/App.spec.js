import App from "../App";
import Row from "../Row";
import { TestHelper, setupGlobalObject } from "../../utils/testUtils";

const defaultProps = {};

const helper = new TestHelper(App, defaultProps);

beforeAll(() => {
  setupGlobalObject();
});

const items = [
  { id: 1, text: "foo", isCompleted: false },
  { id: 2, text: "bar", isCompleted: true }
];

it("renders correctly", () => {
  const wrapper = helper.mountComponent({}, true);
  expect(wrapper).toMatchSnapshot();
});

it("renders correct amount of todos", () => {
  const wrapper = helper.mountComponent({});
  wrapper.setState({ items });
  const todos = wrapper.find(Row);

  expect(todos).toHaveLength(items.length);
});
