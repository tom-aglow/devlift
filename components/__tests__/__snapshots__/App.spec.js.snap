// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "flex-end",
      "paddingTop": 30,
    }
  }
>
  <NewTodo
    inputValue=""
    onAddItem={[Function]}
    onChange={[Function]}
  />
  <SectionList
    data={Array []}
    disableVirtualization={false}
    enableEmptySections={true}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onEndReachedThreshold={2}
    onScroll={[Function]}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "id": 1,
              "isCompleted": false,
              "text": "foo bar",
            },
          ],
          "key": "Personal",
          "title": "Personal",
        },
      ]
    }
    stickySectionHeadersEnabled={true}
    style={
      Object {
        "backgroundColor": "#b0beff",
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
